sources/packages/backend/apps/api/src/services/education-program-offering/custom-validators/has-valid-funded-weeks-for-aviation-offering-credentials.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  ValidatorConstraintInterface,
+  ValidatorConstraint,
+  registerDecorator,
+  ValidationOptions,
+  ValidationArguments,
+} from "class-validator";
+import {
+  MaximumFundedWeeksForAviationOfferingCredentials,
+  OfferingValidationModel,
+  StudyBreak,
+} from "../education-program-offering-validation.models";
+import { OfferingCalculationValidationBaseConstraint } from "./offering-calculation-validation-base-constraint";
+import {
+  AviationCredentialTypeOptions,
+  AviationYesNoOptions,
+} from "@sims/sims-db";
+
+/**
+ * For an offering that contains study breaks and is an aviation offering,
+ * execute the calculation of the funded study period to validate if it matches
+ * the maximum allowed study period amount of funded weeks for the selected
+ * aviation offering credential.
+ */
+@ValidatorConstraint()
+class HasValidFundedWeeksForAviationOfferingCredentialsConstraint
+  extends OfferingCalculationValidationBaseConstraint
+  implements ValidatorConstraintInterface
+{
+  validate(studyBreaks: StudyBreak[], args: ValidationArguments): boolean {
+    const offeringModel = args.object as OfferingValidationModel;
+    if (offeringModel.isAviationOffering === AviationYesNoOptions.No) {
+      return true;
+    }
```

Comment:
This condition can be avoided because when `AviationYesNoOptions.No` then the credential won't be validated.